src/Workspaces/Core/Portable/Classification/SyntaxClassification/AbstractSyntaxClassificationService.Worker.cs

```diff
@@ -98,10 +99,15 @@ private void ProcessNodes()
                 {
                     _cancellationToken.ThrowIfCancellationRequested();
                     var nodeOrToken = _pendingNodes.Pop();
+                    var intervalsThatIntersect = _textSpanIntervalTree.GetIntervalsThatIntersectWith(
+                            nodeOrToken.FullSpan.Start, nodeOrToken.FullSpan.Length);
 
-                    if (nodeOrToken.FullSpan.IntersectsWith(_textSpan))
+                    if (intervalsThatIntersect.Length > 0)
                     {
-                        ClassifyNodeOrToken(nodeOrToken);
+                        foreach (var interval in intervalsThatIntersect)
+                        {
+                            ClassifyNodeOrToken(nodeOrToken, interval);
```

Comment:
>ClassifyNodeOrToken [](http://example.com/codeflow?start=28&length=19)

It looks like there is something a little tricky here. If this works it's way down into ClassifyStructuredTrivia, each iteration of this loop could add the same trivia structure into _pendingNodes.